ovos_persona_server/embeddings.py

```
@@ -0,0 +1,153 @@
+
+    Raises:
+        HTTPException: If the text embeddings plugin fails to load.
+    """
+    global text_embeddings
+    if text_embeddings is None:
+        try:
+            plugin_class = load_text_embeddings_plugin(settings.text_embeddings_plugin)
+            text_embeddings = plugin_class(settings.embeddings_config)
+        except Exception as e:
+            raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                                detail=f"Failed to load text embeddings plugin: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)